x-pack/plugin/identity-provider/src/test/java/org/elasticsearch/xpack/idp/saml/sp/SamlServiceProviderDocumentTests.java

```diff
@@ -121,6 +147,9 @@ private SamlServiceProviderDocument createFullDocument() throws GeneralSecurityE
         doc1.attributeNames.setEmail("urn:" + randomAlphaOfLengthBetween(4, 8) + "." + randomAlphaOfLengthBetween(4, 8));
         doc1.attributeNames.setName("urn:" + randomAlphaOfLengthBetween(4, 8) + "." + randomAlphaOfLengthBetween(4, 8));
         doc1.attributeNames.setRoles("urn:" + randomAlphaOfLengthBetween(4, 8) + "." + randomAlphaOfLengthBetween(4, 8));
+        doc1.attributeNames.setExtensions(
+            randomList(0, 3, () -> "urn:" + randomAlphaOfLengthBetween(4, 8) + "." + randomAlphaOfLengthBetween(4, 8))
```

Comment:
The `testStreamRoundTripWithAllFields` now fails randomly in `assertToXContentEquivalent`. The problem seem to be with the ordering of `extensions` items: 

```
java.lang.AssertionError: Error when comparing xContent.
                           acs: same [https://cEtrLOYX.LakSo/saml/acs]
                    attributes: 
                           email: same [urn:hAVRt.inLOJA]
                      extensions: 
                                 0: expected String [urn:pDeptYdD.rehxgHP] but was String [urn:BeZmekTl.QhNbhUns]
                                 1: expected String [urn:BeZmekTl.QhNbhUns] but was String [urn:pDeptYdD.rehxgHP]
```

Reply:
I think I'll just make it a single element then.